### PR TITLE
Deprecate EmrContainerOperatorAsync

### PR DESCRIPTION
**Pull request by Lee-W**
# What's changed
Deprecate `EmrContainerOperatorAsync` and fallback to `EmrContainerOperator` with `deferrable=True`
This is blocked by the release of ~~https://github.com/apache/airflow/pull/36417~~ https://github.com/apache/airflow/issues/36467

upgrade minimum apache-airflow-providers-amazon to 8.15.0

# Why this change
The logic of  `EmrContainerOperatorAsync` has been contributed back to OSS airflow as part of `EmrContainerOperator`

### EmrContainerOperatorAsync
https://github.com/astronomer/astronomer-providers/blob/b5a9417919d8a13b8e519d2582ce705c4689b7c6/astronomer/providers/amazon/aws/sensors/emr.py#L38-L65

### EmrContainerOperator
https://github.com/apache/airflow/blob/999b70178a1f5d891fd2c88af4831a4ba4c2cbc9/airflow/providers/amazon/aws/sensors/emr.py#L318-L346

### CODE REVIEW

**Review by pankajastro — APPROVED** [review]
LGTM

File: astronomer/providers/amazon/aws/operators/emr.py [hunk 1]
```diff
@@ -1,95 +1,22 @@
...
-        Callback for when the trigger fires - returns immediately.
-        Relies on trigger to throw an exception, otherwise it assumes execution was
-        successful.
-        """
-        if "status" in event and event["status"] == "error":
-            raise AirflowException(event["message"])
-        self.log.info(event["message"])
-        job_id: str = event["job_id"]
-        return job_id
+        return super().__init__(*args, deferrable=True, **kwargs)
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/emr.py:22]: Do we need a return here?

Comment (Lee-W) [reply astronomer/providers/amazon/aws/operators/emr.py:22]: Nope, i think it's accidentally added. let me send a quick fix

Comment (Lee-W) [reply astronomer/providers/amazon/aws/operators/emr.py:22]: https://github.com/astronomer/astronomer-providers/pull/1430

**Pull request merged**